checkout/interfaces/controller/checkoutcontroller.go

```diff
@@ -244,26 +245,30 @@ func (cc *CheckoutController) PlaceOrderAction(ctx context.Context, r *web.Reque
 		return cc.responder.Render("checkout/emptycart", nil).SetNoCache()
 	}
 
-	err := cc.orderService.SetSources(ctx, session)
+	err = cc.orderService.SetSources(ctx, session)
 	if err != nil {
 		cc.logger.Error("OnStepCurrentCartPlaceOrder SetSources Error ", err)
 		return cc.responder.Render("checkout/carterror", nil).SetNoCache()
 	}
+	var placedOrderInfo *application.PlaceOrderInfo
+	if cc.orderService.HasLastPlacedOrder(ctx) {
+		placedOrderInfo, _ = cc.lastPlaceOrderInfo(ctx)
+		cc.orderService.ClearLastPlacedOrder(ctx)
+	} else {
+		placedOrderInfo, err = cc.orderService.CurrentCartPlaceOrderWithPaymentProcessing(ctx, session)
 
-	//No Payment selected:	return cc.showCheckoutFormWithErrors(ctx, r, *decoratedCart, nil, errors.New("No payment selected"))
-	placedOrderInfo, err := cc.orderService.CurrentCartPlaceOrderWithPaymentProcessing(ctx, session)
-
-	if err != nil {
-		cc.logger.WithContext(ctx).WithField("subcategory", "checkoutError").WithField("errorMsg", err.Error()).Error(fmt.Sprintf("place order failed: cart id: %v / total-amount: %v", decoratedCart.Cart.EntityID, decoratedCart.Cart.GrandTotal()))
-		if paymentError, ok := err.(*paymentDomain.Error); ok {
-			if paymentError.ErrorCode == paymentDomain.ErrorPaymentAbortedByCustomer {
-				return cc.showCheckoutFormAndHandleSubmit(ctx, r, "checkout/checkout")
-			}
-			if cc.showReviewStepAfterPaymentError && !cc.skipReviewAction {
-				return cc.showReviewFormWithErrors(ctx, *decoratedCart, paymentError)
+		if err != nil {
```

Comment:
I think you can just have the assignment of the placeOrderInfo in the if else 
and move the err evaluation up.
If we also assign the err for lastPlaceOrderInfo it is always evaluated and logged in case of an error, regardless what kind of error

The fetch of the placeOrderInfo could then also be moved to a private function